pkg/pulumiyaml/run.go

```diff
@@ -366,6 +378,8 @@ type Runner struct {
 
 	sdiags syncDiags
 
+	intermediatesSet bool
```

Comment:
This field is not strictly necessary, since we can test `intermediates == nil`. I believe thats what we did in the past.